server/hotrod/src/main/java/org/infinispan/server/hotrod/Encoder2x.java

```diff
@@ -692,6 +730,11 @@ private Optional<AbstractTopologyResponse> generateTopologyResponse(short client
          }
       }
 
+      if (topologyId != 1) {
+         topologyId = cacheTopology.hashCode();
```

Comment:
`writeHashTopologyUpdate()`  sends the `ConsistentHash ch = cacheTopology.getReadConsistentHash();`. I think it's `hashcode()` should be used instead to minimize updating the client too often.

Reply:
I believe it is pretty minimal savings as there is only one stage of rehash that doesn't change the read consistent hash afaik, in the middle. However, it will have some benefit and hashCode is less work. I can change it.